src/django/api/processing.py

```diff
@@ -33,7 +36,53 @@ def _report_error_to_rollbar(file, request):
                 'file_name': file.name})
 
 
-def get_excel_sheet(file, request):
+def get_xlsx_sheet(file, request):
+    import defusedxml
+    from defusedxml.common import EntitiesForbidden
+
+    defusedxml.defuse_stdlib()
+
+    try:
+        wb = load_workbook(filename=file)
+        ws = wb[wb.sheetnames[0]]
+
+        return ws
+
+    except EntitiesForbidden:
+        _report_error_to_rollbar(file, request)
+        raise ValidationError('This file may be damaged and '
+                              'cannot be processed safely')
+
+
+def parse_xlsx(file, request):
+    try:
+        ws = get_xlsx_sheet(file, request)
+
+        percent_col = []
+
+        for cell in ws[2]:
+            if '%' in cell.number_format:
+                percent_col.append(get_column_letter(cell.column))
+
+        if percent_col:
+            for col in percent_col:
+                for cell in ws[col]:
+                    if cell.row != 1:
+                        cell.value = f"{cell.value * 100}%"
+
+        header = ','.join([cell.value for cell in ws[1]])
+
+        rows = ['"{}"'.format(
+            '","'.join([str(cell.value) for cell in ws[idx]]))
+                for idx in range(2, ws.max_row)]
```

Comment:
When I processed the `XLSX.-.Testing.xlsx` worksheet I only ended up with 18 of the 18 roes. I suspect this is becasue Python's `range` function does not include the last value

```python
>>> list(range(1,10))
[1, 2, 3, 4, 5, 6, 7, 8, 9]
```
I expect this is a simple fix
```python
for idx in range(2, ws.max_row + 1)]
```